converter/src/main/java/gov/cms/qpp/conversion/model/error/Detail.java

```diff
@@ -31,42 +39,51 @@ public Detail() {
 	 * Copy constructor
 	 */
 	public Detail(Detail detail) {
-		this(detail.getMessage(), detail.getPath(), detail.getValue(), detail.getType());
+		setErrorCode(detail.getErrorCode());
+		setMessage(detail.getMessage());
+		setPath(detail.getPath());
+		setValue(detail.getValue());
+		setType(detail.getType());
 	}
 
 	/**
-	 * Constructs a {@code Detail} with just a description.
-	 *
-	 * @param text A description of the error.
+	 * Creates a mutable Detail based on the given error and node
 	 */
-	public Detail(String text) {
-		this.message = text;
+	public static Detail forErrorAndNode(LocalizedError error, Node node) {
+		Objects.requireNonNull(node, "node");
+
+		Detail detail = forErrorCode(error);
+		detail.setPath(node.getPath());
+		return detail;
 	}
 
 	/**
-	 * Constructs a {@code Detail} with a description and an path to point where the error is in the original document.
-	 *
-	 * @param text A description of the error.
-	 * @param path A path to where the error is.
+	 * Creates a mutable Detail based on the given error
+	 * @param code
+	 * @return
 	 */
-	public Detail(String text, String path) {
-		this.message = text;
-		this.path = path;
+	public static Detail forErrorCode(LocalizedError error) {
+		Objects.requireNonNull(error, "error");
+
+		Detail detail = new Detail();
+		detail.setErrorCode(error.getErrorCode());
+		detail.setMessage(error.getMessage());
+		return detail;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "setType" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1699)